pulser-core/pulser/backend/config.py

```diff
@@ -14,28 +14,236 @@
+            model, this option signals the backend to prefer it over the noise
+            model given with this configuration.
+        noise_model: An optional noise model to emulate the sequence with.
+            Ignored if the sequence's device has default noise model and
+            `prefer_device_noise_model=True`.
     """
 
-    backend_options: dict[str, Any] = field(default_factory=dict)
+    observables: Sequence[Observable]
+    default_evaluation_times: np.ndarray | Literal["Full"]
+    initial_state: StateType | None
+    with_modulation: bool
+    interaction_matrix: pm.AbstractArray | None
+    prefer_device_noise_model: bool
+    noise_model: NoiseModel
+
+    def __init__(
+        self,
+        *,
+        observables: Sequence[Observable] = (),
+        # Default evaluation times for observables that don't specify one
+        default_evaluation_times: Sequence[SupportsFloat] | Literal["Full"] = (
+            1.0,
+        ),
+        initial_state: StateType | None = None,  # Default is ggg...
+        with_modulation: bool = False,
+        interaction_matrix: ArrayLike | None = None,
+        prefer_device_noise_model: bool = False,
+        noise_model: NoiseModel = NoiseModel(),
+        **backend_options: Any,
+    ) -> None:
+        """Initializes the EmulationConfig."""
+        obs_tags = []
+        for obs in observables:
+            if not isinstance(obs, Observable):
+                raise TypeError(
+                    "All entries in 'observables' must be instances of "
+                    f"Observable. Instead, got instance of type {type(obs)}."
+                )
+            obs_tags.append(obs.tag)
+        repeated_tags = [k for k, v in Counter(obs_tags).items() if v > 1]
+        if repeated_tags:
+            raise ValueError(
+                "Some of the provided 'observables' share identical tags. Use "
+                "'tag_suffix' when instantiating multiple instances of the "
+                "same observable so they can be distinguished. "
+                f"Repeated tags found: {repeated_tags}"
+            )
 
+        if default_evaluation_times != "Full":
+            eval_times_arr = np.array(default_evaluation_times, dtype=float)
+            if np.any((eval_times_arr < 0.0) | (eval_times_arr > 1.0)):
+                raise ValueError(
+                    "All evaluation times must be between 0. and 1. "
+                    f"Instead, got {default_evaluation_times}."
+                )
+            default_evaluation_times = cast(Sequence[float], eval_times_arr)
+
+        if initial_state is not None and not isinstance(initial_state, State):
+            raise TypeError(
+                "When defined, 'initial_state' must be an instance of State;"
+                f" got object of type {type(initial_state)} instead."
+            )
+
+        if interaction_matrix is not None:
+            interaction_matrix = pm.AbstractArray(interaction_matrix)
```

Comment:
should we symmetrize this? or at least check for symmetry?

Reply:
Yes, I'll add a check